AI_ML/REGRESSION/LEAST_SQUARES/LEAST_SQUARES.py

```diff
@@ -20,22 +23,13 @@ def LEAST_SQUARES(dc_inputs: list[DataContainer], params: dict) -> DataContainer
         y: fitted line computed with returned regression weights
     """
 
-    x: np.ndarray = np.eye(3)
-    y: np.ndarray = np.eye(3)
-    a: np.ndarray = np.eye(3)
-
-    if len(dc_inputs) == 0 or len(dc_inputs) > 2:
-        raise ValueError(
-            f"To compute least squares, LEAST_SQUARES node requires one or two inputs, {len(dc_inputs)} was given!"
-        )
-    if len(dc_inputs) == 1 and dc_inputs[0].type == "ordered_pair":
-        if len(np.array(dc_inputs[0].y)) != 0:
-            x = np.array(dc_inputs[0].y)
-            y = np.array(dc_inputs[0].y)
+    if b is None and isinstance(a, OrderedPair):
+        if (len(a.y)) != 0:
+            x = a.y
```

Comment:
Is this a typo?

Reply:
It was a check to cover cases when it was receiving inputs directly from linspace or not

Reply:
This just always overwrites the x values with the y values, giving the same line each time, so it should be deleted
Either way, we should probably have Linspace return a Vector instead of an OrderedPair, since having `None` as the x value is a bit misleading.
This should actually be `if not a.x` though.

Reply:
so do I just keep what's in the `else` statement?

Reply:
Yes, and you can delete the if condition (I just opened a PR to add a vector type so you don't need to handle this case anymore)